src/core/core.controller.js

```diff
@@ -291,6 +297,14 @@ module.exports = function(Chart) {
 					me.scale = scale;
 				}
 			});
+			// clear up discarded scales
+			helpers.each(updated, function(hasUpdated, id) {
```

Comment:
@etimberg last commit here was definitely a mistake. Fixed here.
So now whenever user specify a scale with a new `id` or the `type` is changed, user lose the reference to the old scale.